app/controllers/home.js

```diff
@@ -24,6 +77,37 @@ export default Ember.Controller.extend({
             if(this.get('model.theme').get('canRedo')) {
                 this.get('model.theme').redo();
             }
+        },
+        publish(){//publish will take unpublished data and put it in to page data on firebase 
+            this.set('published' , true)
+            this.send('save' , this.get('model.guid') )
+            this.store.findRecord('home', this.get('model.guid')).then((data)=> {               
+                data.set('pageData',  data.get('unpublishedPageData'));
+                data.save();
+            });
+            setTimeout(()=>{
```

Comment:
because the save/publish happens too fast for the text to even appear and it is used for user feedback so they know it saved